internal/state/configuration.go

```diff
@@ -107,25 +110,26 @@ func (b *configBuilder) upsertListener(l *listener) {
 
 func (b *configBuilder) build() Configuration {
 	return Configuration{
-		HTTPServers: b.http.build(),
-		SSLServers:  b.ssl.build(),
+		HTTPServers: b.http.buildHTTP(),
```

Comment:
```
		HTTPServers: b.http.buildHTTP(),
		SSLServers:  b.ssl.buildSSL(),
```

Since we're using different methods (interfaces), It seems that  now we have two types -one that deals with SSL and one deal with HTTP - inside `virtualServerBuilder `. I think this makes the code more difficult to comprehend and use.

I wonder if we can still use `build()` interface, so virtualServerBuilder hides the details  for us whether we deal with HTTP or HTTPS. Perhaps that could be done via some flag in virtualServerBuilder ?

This is not critical. I wonder if you had similar thoughts.

Reply:
Yeah I did have similar thoughts...Let me tinker with it and see if I can come up with something better.

Reply:
Updated virtual server builder. Let me know what you think